### PR TITLE
Make Prometheus SSL optional

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this is an improvement

Based on feedback we got that running without SSL is preferred in some non-public VPC environments.

This moves most of the YAML generation from the template into the main Terraform code, as recommended by Terraform due to better syntax/type validation.
 
## Checklist

 - [X] Link to issue this PR refers to (if applicable): 
 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: aws/scripts/cloud-init-cratedb-rpm.tftpl [hunk 1]
```diff
@@ -45,8 +45,8 @@ write_files:
       #!/bin/bash
 
       sleep 30
-      curl -sS -H 'Content-Type: application/json' -k -X POST 'http${crate_ssl_enable ? "s" : ""}://localhost:4200/_sql' -d '{"stmt":"CREATE USER ${crate_user} WITH (password = '\''${crate_pass}'\'');"}'
-      curl -sS -H 'Content-Type: application/json' -k -X POST 'http${crate_ssl_enable ? "s" : ""}://localhost:4200/_sql' -d '{"stmt":"GRANT ALL PRIVILEGES TO ${crate_user};"}'
+      curl -sS -H 'Content-Type: application/json' -k -X POST 'http${crate_ssl_enable ? "s" : ""}://localhost:4200/_sql' -d '{"stmt": "CREATE USER ${crate_user} WITH (password = '\''${crate_pass}'\'');"}'
```
Comment (hlcianfagna) [inline aws/scripts/cloud-init-cratedb-rpm.tftpl:48]: Do we need to escape quotes here or modify ec2.tf to exclude quotes from the possible characters for random passwords?

Comment (hammerhead) [reply aws/scripts/cloud-init-cratedb-rpm.tftpl]: The randomly generated password won't include quotes, special characters are already restricted:

https://github.com/crate/crate-terraform/blob/bf2487536e8aa47858e4b238f7423a8ce0bbc52b/aws/ec2.tf#L5-L9

However, we allow providing a custom password. What do you think of f261cf8? Unfortunately, Terrafom messes up the double quote by not escaping it when passing the password to the bash script, so I disallowed it. The bash script itself would be able to handle it (by delegating some of the escaping complexity to jq). Double dollar signs also wouldn't not be allowed with this change.

Comment (hlcianfagna) [reply aws/scripts/cloud-init-cratedb-rpm.tftpl]: This looks good to me, thank you.

**Review by hlcianfagna — APPROVED** [review]

**Comment by hammerhead** [comment]
I added automated [Terraform tests](https://developer.hashicorp.com/terraform/language/tests) now, at least as far as input variable validation goes.

**Pull request merged**